beanie/odm/cache.py

```diff
@@ -40,6 +61,75 @@ def set(self, key, value) -> None:
                 self.cache.popitem(last=False)
         self.cache[key] = CachedItem(value=value)
 
+    def invalidate(self) -> None:
+        self.cache.clear()
+
+
+class RedisCache(Cache):
+    def __init__(
+        self,
+        redis_client,
+        ttl: Optional[Union[int, timedelta]] = None,
+        key_prefix: str = "cache",
+        serializer=pickle,
+    ):
+        self.redis_client = redis_client
+        self.key_prefix = key_prefix
+        self.ttl = self._normalize_ttl(ttl)
+        self.serializer = serializer
+
+    def _normalize_object_ids(self, value):
+        if isinstance(value, dict):
+            if "_id" in value:
+                value["_id"] = str(value["_id"])
+        elif isinstance(value, Sequence):
+            for row in value:
+                self._normalize_object_ids(row)
+        return value
+
+    def _serialize(self, value):
+        if not value:
+            return None
+        if self.serializer is not pickle:
+            # Turn ObjectId to string for those serializer that can't
+            # serialize ObjectId, like: msgpack and json
+            value = self._normalize_object_ids(value)
+        return self.serializer.dumps(value)
+
+    def _deserialize(self, value):
+        try:
+            loaded = self.serializer.loads(value)
+            return loaded
+        except (
+            TypeError,
+            pickle.UnpicklingError,
+            UnicodeDecodeError,
+            msgpack.exceptions.UnpackException,
+            msgpack.exceptions.ExtraData,
+        ):
+            return None
```

Comment:
I didn't get the logic here. I think, users should know, that the cache is broken. May be I don't see something